### PR TITLE
nrf_rpc: Add test for ot_message allocations.

### PR DESCRIPTION
**Pull request by canisLupus1313**
Commit adds unit tests for ot message manipulation in nrf rpc Openthread part.

### CODE REVIEW

File: subsys/net/openthread/rpc/server/ot_rpc_udp.c [hunk 1]
```diff
@@ -104,6 +104,8 @@ static void ot_rpc_udp_open(const struct nrf_rpc_group *group, struct nrf_rpc_cb
 		return;
 	}
 
+	printf("dupa\n");
```
Comment (alxelax) [inline subsys/net/openthread/rpc/server/ot_rpc_udp.c:107]: I guess this is the new test result status? :)

File: subsys/net/openthread/rpc/Kconfig [hunk 2]
```diff
@@ -59,6 +59,18 @@ config OPENTHREAD_RPC_CLIENT_NUM_SENT_COAP_REQUESTS
 
 endmenu # "OpenThread over RPC client configuration"
 
+menu "OpenThread over RPC client configuration"
+	depends on OPENTHREAD_RPC_SERVER
+
+	config OPENTHREAD_RPC_MESSAGE_POOL
```
Comment (Damian-Nordic) [inline subsys/net/openthread/rpc/Kconfig:65]: Could you not indent options in `menu` to be consistent with the rest of NCS?

File: subsys/net/openthread/rpc/Kconfig [hunk 3]
```diff
@@ -59,6 +59,18 @@ config OPENTHREAD_RPC_CLIENT_NUM_SENT_COAP_REQUESTS
...
+		int "Maximum number of messages that can be allocated on server"
+		default 8
+		help
+		  Defines maximum number of messages that can be allocated on server at
```
Comment (Damian-Nordic) [inline subsys/net/openthread/rpc/Kconfig:69]: In fact, this is the maximum number of otMessage keys/handles that are passed to the RPC client and you can allocate many more otMessages on the server. Can this be rephrased to make it clear?

File: tests/subsys/net/openthread/rpc/server/src/message_suite.c [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+ZTEST(ot_rpc_message, test_otUdpNewMessage_failing)
+{
+	otUdpNewMessage_fake.return_val = NULL;
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(0), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_NULL));
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(0), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_TRUE, CBOR_UINT8(40)));
```
Comment (Damian-Nordic) [inline tests/subsys/net/openthread/rpc/server/src/message_suite.c:52]: You don't verify if these `true` and 40 are decoded properly. Can it be added?

File: tests/subsys/net/openthread/rpc/server/src/message_suite.c [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(1), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_NULL));
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(2), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_UDP_NEW_MESSAGE, CBOR_TRUE, CBOR_UINT8(40)));
+
+	zassert_equal(otUdpNewMessage_fake.call_count, 2);
+
+	mock_nrf_rpc_tr_expect_add(RPC_RSP(), NO_RSP);
+	mock_nrf_rpc_tr_receive(RPC_CMD(OT_RPC_CMD_MESSAGE_FREE, 1));
```
Comment (Damian-Nordic) [inline tests/subsys/net/openthread/rpc/server/src/message_suite.c:73]: Would be great to verify that `otMessageFree` has received the right pointer.

**Review by Damian-Nordic — APPROVED** [review]
Some nits and suggestions but lgtm otherwise.

File: subsys/net/openthread/rpc/Kconfig (hunk 2 above)
Comment (canisLupus1313) [reply subsys/net/openthread/rpc/Kconfig]: fixed.

File: tests/subsys/net/openthread/rpc/server/src/message_suite.c (hunk 4 above)
Comment (canisLupus1313) [reply tests/subsys/net/openthread/rpc/server/src/message_suite.c]: fixed.

File: tests/subsys/net/openthread/rpc/server/src/message_suite.c (hunk 5 above)
Comment (canisLupus1313) [reply tests/subsys/net/openthread/rpc/server/src/message_suite.c:85]: fixed.

File: subsys/net/openthread/rpc/Kconfig [hunk 6]
```diff
@@ -59,6 +59,18 @@ config OPENTHREAD_RPC_CLIENT_NUM_SENT_COAP_REQUESTS
...
 endmenu # "OpenThread over RPC client configuration"
 
+menu "OpenThread over RPC client configuration"
+	depends on OPENTHREAD_RPC_SERVER
+
+config OPENTHREAD_RPC_MESSAGE_POOL
+	int "Maximum number of messages key identifiers that can be allocated on server"
+	default 8
+	help
+		Defines maximum number of messages key indentifiers that can be allocated on server at
```
Comment (alxelax) [inline subsys/net/openthread/rpc/Kconfig:69]: Indents should be aligned with others.

File: subsys/net/openthread/rpc/server/ot_rpc_message.c [hunk 7]
```diff
@@ -118,9 +122,9 @@ static void ot_rpc_msg_udp_new(const struct nrf_rpc_group *group, struct nrf_rpc
 	ot_msg_key key;
 	struct nrf_rpc_cbor_ctx rsp_ctx;
 	otMessageSettings settings;
-	otMessageSettings *pSettings;
+	otMessageSettings *pSettings = &settings;
```
Comment (alxelax) [inline subsys/net/openthread/rpc/server/ot_rpc_message.c:125]:
```suggestion
	otMessageSettings *p_settings = &settings;
```
No reason in OpenThread coding stile usage in rpc

**Review by alxelax — COMMENTED** [review]
LGTM just two minor comments

File: subsys/net/openthread/rpc/Kconfig (hunk 6 above)
Comment (canisLupus1313) [reply subsys/net/openthread/rpc/Kconfig]: fixed.

File: subsys/net/openthread/rpc/server/ot_rpc_message.c (hunk 7 above)
Comment (canisLupus1313) [reply subsys/net/openthread/rpc/server/ot_rpc_message.c]: fixed.

**Review by nordicjm — APPROVED** [review]

**Pull request merged**